projects/packages/my-jetpack/_inc/components/welcome-flow/index.tsx

```diff
@@ -37,29 +39,53 @@ const WelcomeFlow: FC = () => {
 		return 'evaluation-processing';
 	}, [ isProcessingEvaluation, siteIsRegistered ] );
 
+	useEffect( () => {
+		if ( prevStep !== currentStep ) {
+			recordEvent( 'jetpack_myjetpack_welcome_banner_step_change', { currentStep } );
+			setPrevStep( currentStep );
+		}
+	}, [ currentStep, prevStep, recordEvent ] );
+
 	const onDismissClick = useCallback( () => {
 		recordEvent( 'jetpack_myjetpack_welcome_banner_dismiss_click', {
 			currentStep,
 			isUserConnected,
 			isSiteConnected,
 		} );
-		setVisible( false );
 		dismissWelcomeBanner();
 	}, [ recordEvent, currentStep, isUserConnected, isSiteConnected, dismissWelcomeBanner ] );
 
 	const handleEvaluation = useCallback(
-		( _values: { [ key in EvaluationAreas ]: boolean } ) => {
+		( values: { [ key in EvaluationAreas ]: boolean } ) => {
+			const goals = Object.keys( values ).filter( key => values[ key ] );
+
 			setIsProcessingEvaluation( true );
-			setTimeout( () => {
-				// TODO: Mock "evaluation": Implement the evaluation endpoint
-				setVisible( false );
-				dismissWelcomeBanner();
-			}, 3_000 );
+			recordEvent( 'jetpack_myjetpack_welcome_banner_submit_evaluation', { goals } );
+			submitEvaluation(
+				{ queryParams: { goals } },
+				{
+					onSuccess: recommendations => {
+						recordEvent( 'jetpack_myjetpack_welcome_banner_evaluation_success', {
+							recommendations,
```

Comment:
I think we'll need to extract the values from these recommendations and concatenate them together, I'm seeing this in the tracks
![image](https://github.com/Automattic/jetpack/assets/65001528/b7608162-0a74-407e-ae44-f83813bffbd7)

Reply:
That's fixed, thanks! Sorry for that overlook.